pkg/build/controller/controller.go

```diff
@@ -288,6 +290,20 @@ func (bc *BuildPodController) HandlePod(pod *kapi.Pod) error {
 	case kapi.PodRunning:
 		// The pod's still running
 		nextStatus = buildapi.BuildPhaseRunning
+	case kapi.PodPending:
+		if build.Status.Phase == buildapi.BuildPhasePending {
+			// When build is in pending state due to missing pull secrets for the image
+			// indicate this condition in reason.
+			if secret := build.Spec.Output.PushSecret; secret != nil {
+				if bc.SecretClient.Secrets(build.Namespace).Get(secret.Name); err != nil && errors.IsNotFound(err) {
```

Comment:
how often is this going to get called?  you should probably not fetch the secrets if the reason is already set to status missing secret.